pkg/aws/vpc/reconciler.go

```diff
@@ -0,0 +1,88 @@
+package vpc
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/aws-vpc-operator/pkg/errors"
+)
+
+const (
+	defaultVPCCidr = "10.0.0.0/16"
+)
+
+type Spec struct {
+	ClusterName string
+	RoleARN     string
+	VpcId       string
+	CidrBlock   string
+}
+
+type Status struct {
+	VpcId     string
+	CidrBlock string
+	Tags      map[string]string
+}
+
+type Reconciler interface {
+	Reconcile(ctx context.Context, spec Spec) (Status, error)
+}
+
+func NewReconciler(client Client) (Reconciler, error) {
+	if client == nil {
+		return nil, microerror.Maskf(errors.InvalidConfigError, "client must not be empty")
+	}
+
+	return &reconciler{
+		client: client,
+	}, nil
+}
+
+type reconciler struct {
+	client Client
+}
+
+func (s *reconciler) Reconcile(ctx context.Context, spec Spec) (Status, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Started reconciling VPC")
```

Comment:
Let's add the VPC or cluster name to the logs.

Reply:
+1

Reply:
ideally add  keys  once you get teh cluster name and than just propagate the new logger with the key values of cluster

Reply:
`AWSCluster` CR namespace and name is already set there, even twice, not sure how/where exactly (I assume by kubebuilder/logr).

It looks like this:
```
INFO	Started reconciling VPC	{"controller": "awscluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "AWSCluster", "AWSCluster": {"name":"nikola0006","namespace":"org-giantswarm"}, "namespace": "org-giantswarm", "name": "nikola0006", "reconcileID": "48e35810-d642-4a0f-a858-5df90e39644a"}
```

There is actually a line where I log it explicitly (forgot to remove that), it looks like this (namespace and name are written 3 times):
```
INFO	Started reconciling AWSCluster	{"controller": "awscluster", "controllerGroup": "infrastructure.cluster.x-k8s.io", "controllerKind": "AWSCluster", "AWSCluster": {"name":"nikola0006","namespace":"org-giantswarm"}, "namespace": "org-giantswarm", "name": "nikola0006", "reconcileID": "48e35810-d642-4a0f-a858-5df90e39644a", "namespace": "org-giantswarm", "name": "nikola0006"}
```